wagtail/wagtailcore/models.py

```diff
@@ -367,6 +369,9 @@ def __init__(self, *args, **kwargs):
             # set content type to correctly represent the model class that this was
             # created as
             self.content_type = ContentType.objects.get_for_model(self)
+        if 'show_in_menus' not in kwargs:
+            # if the value is not set on submit, refer to the model's setting
+            self.show_in_menus = self.show_in_menus_default
```

Comment:
The broken tests are probably caused by these line. Try replacing all the method with the following and run the test suite one your computer to check:
```python
    def __init__(self, *args, **kwargs):
        super(Page, self).__init__(*args, **kwargs)
        if self.pk is None:
            if self.content_type_id is None:
                # this model is being newly created rather than retrieved from the db;
                # set content type to correctly represent the model class that this was
                # created as
                self.content_type = ContentType.objects.get_for_model(self)
            if 'show_in_menus' not in kwargs:
                # if the value is not set on submit, refer to the model's setting
                self.show_in_menus = self.show_in_menus_default
```